### ISSUE TITLE
1837 Coal company direct revenue does or does not work?

### ISSUE THREAD

**Issue by neutronc** [Rails-18xx/Rails#369]
The revenue of the EPP should be 40 to company and 10 to player but its split evenly.
This is from the latest commit in this repo.
[1837_20210323_2111_Marcus.zip](https://github.com/Rails-18xx/Rails/files/6192762/1837_20210323_2111_Marcus.zip)

**Comment by erik-vos** [comment]
I had to checkout an earlier commit to get this file loaded, but then, when I undo to the EPP turn and redo the revenue step, the split is exactly what it should be.  I haven't installed 2.4.5 yet, will try that later.

**Comment by erik-vos** [comment]
Strange. The above I have run from commit 6ca9c638.
Just installed 2.4.5, loaded your save file, EPP shows wrong numbers, undo, and then comes the surprise: if I redo, it remains wrong, but if I press SetRevenue myself, the numbers suddenly are correct. Miracles....

**Comment by erik-vos** [comment]
Here is the same file, with the correct values, as it came out of my run, as I described above.

[1837_20210323_2111_MarcusBetter.zip](https://github.com/Rails-18xx/Rails/files/6192989/1837_20210323_2111_MarcusBetter.zip)

Are you sure you have not been loading (part of) an older saved file?

**Comment by erik-vos** [comment]
It very much looks like some local setting. So I checked my local profile, but did not find anything that by name might relate to this problem. But I do remember that to make my direct revenue kludge working, I had to tweak something somewhere. Or it is a side effect of some other setting.

Anyway, it clearly it time to revisit the direct revenue code, as I announced would be necessary at some point.

**Comment by erik-vos** [comment]
See recent update in #346. It appears that the problem still surfaces in Ubuntu, whereas it runs fine in Windows 10. 
Something wrong with the Linux version?

**Comment by neutronc** [comment]
I downloaded the 2.4.6 windows executable and i got the wrong result:
![grafik](https://user-images.githubusercontent.com/6752457/113877938-71dd3b00-97b9-11eb-8ee1-8029f2845114.png)

Coal Mine revenue was recognised as 20 K and should have gone to Company Treasury but instead it was added to dividend 20+10 and then split. Correct would have been 25 for company and 5 for player x.

[1837_20210407_1351_X.zip](https://github.com/Rails-18xx/Rails/files/6272081/1837_20210407_1351_X.zip)

**Comment by neutronc** [comment]
downloaded 2.4.6.deb and got the same result Dividend should have been 30 but was calculated as 60 (sum of 30 Coal income and 30 station income) thus was split 30 for player and 30 for company what should have been 45 for company and 15 for player.

**Comment by neutronc** [comment]
I think the problem is that you have to set an option at starting a new game of 1837...

**Comment by erik-vos** [comment]
Not an option but a profile setting it turns out to be. 
When this issue came up, that was the first place I looked, because it seemed to be player dependent. But I could not find anything.

Today I got the brainwave to look at startup under the Configuration button, where under the Appearance tab is an item "Show spinner in revenue step".  I have it always on, and when I just set it off, I suddenly could reproduce your results. So I bet John and Martin have it off!

So my advice is: set it on. And now I know where to look to fix this problem. That profile option must have unwanted side effects.
